### PR TITLE
Update /api/v1/files/{storage}/{path} endpoint

### PR DESCRIPTION
**Pull request by Majkluss**

### CODE REVIEW

File: spec/openapi.yaml [hunk 1]
```diff
@@ -624,137 +611,186 @@ components:
...
+
+    OtherFileInfo:
+      $ref: '#/components/schemas/FileTreeObject'
+
+    FolderInfo:
+      allOf:
+        - $ref: '#/components/schemas/FileTreeObject'
+        - type: object
+          properties:
+            file_count:
```
Comment (ondratu) [inline spec/openapi.yaml:790]: I'm not sure about that information on Buddy FW. Please check it with @vorner.

File: spec/openapi.yaml [hunk 2]
```diff
@@ -624,137 +611,188 @@ components:
...
+          example: "Spice_Harvester_0.3mm_PLA_MK3S_12m.gcode"
         display_name:
           type: string
           description: Long Filename
           example: "Spice_Harvester_0.3mm_PLA_MK3S_12m.gcode"
-        creation_timestamp:
+        path:
+          type: string
+          example: "/local/Dune/Spice_Harvester_0.3mm_PLA_MK3S_12m.gcode"
+        display_path:
```
Comment (ondratu) [inline spec/openapi.yaml:627]: We have to discuss that one more time in our internal Slack channel. It bad, if one path contains file_name and another not.

File: spec/openapi.yaml [hunk 3]
```diff
@@ -624,137 +611,186 @@ components:
...
-            max_exposure_time:
-              type: string
-              description: Seconds
-              example: "3"
-            max_initial_exposure_time:
-              type: string
-              description: Seconds
-              example: "3"
-            min_exposure_time:
+            download:
```
[the remark this answers is not in the document]
Comment (Majkluss) [reply spec/openapi.yaml:652]: True, this should be under PRINT_FILE type only. I'll fix it.

File: spec/openapi.yaml (hunk 2 above)
Comment (Majkluss) [reply spec/openapi.yaml:627]: I think it's okay to remove filename from the both paths, so we'll have `/local/Dune` and `/PrusaLink gcodes/Dune`

File: spec/openapi.yaml [hunk 4]
```diff
@@ -624,137 +611,190 @@ components:
           type: boolean
           example: true
 
-    FileInfo:
+    FileTreeObject:
       type: object
       properties:
-        type:
+        origin:
```
Comment (ondratu) [inline spec/openapi.yaml:617]: What usage of origin should be?

**Review by ondratu — CHANGES_REQUESTED** [review]

File: spec/openapi.yaml (hunk 4 above)
Comment (Majkluss) [reply spec/openapi.yaml]: Removed :)

**Review by ondratu — APPROVED** [review]

File: spec/openapi.yaml [hunk 5]
```diff
@@ -626,135 +645,190 @@ components:
...
-            min_initial_exposure_time:
-              type: string
-              description: Seconds
-              example: "1"
-            nozzle_diameter:
-              type: string
-              description: Milimeters
-              example: "0.4"
-            print_time:
+              example: "/PrusaLink gcodes/Dune"
```
Comment (vorner) [inline spec/openapi.yaml:694]: Are we expected to provide both long filename and _decode the whole prefix_ too, for each file? :-O

File: spec/openapi.yaml (hunk 1 above)
Comment (vorner) [reply spec/openapi.yaml:824]: Well, considering we have the array with children, isn't this a bit redundant? But if you really need us to count them, we can do so, but will send the field _after_ the `children` array 😈

File: spec/openapi.yaml [hunk 6]
```diff
@@ -626,135 +645,190 @@ components:
 
     FileInfo:
       type: object
+      required:
+        - name
+        - path
+        - ro
+        - type
+        - c_timestamp
+        - m_timestamp
```
Comment (vorner) [inline spec/openapi.yaml:654]: I'm not sure we have both `c_timestamp` and `m_timestamp` available (might need to check that out). Why are both mandatory?

**Comment by Majkluss** [comment]
> @vorner
> 
> > Well, considering we have the array with children, isn't this a bit redundant? But if you really need us to count them, we can do so, but will send the field after the children array smiling_imp
> 
> I would love to have there info about files and folders inside of a folder and I don't need its children.
> 
> @ondratu @Majkluss
> 
> Endpoint `GET /api/v1/files/${storage}/${path}/` is totally useless and this is bad. I would suggest to change it so it returns list of files and folders if it is being used for folders, so we can get rid of `/api/files/path/${storage}/${path}` endpoint.
> 
> It seem very logical to me. What do you think?

This endpoint should replace the above mentioned one, so, yes :)

File: spec/openapi.yaml (hunk 5 above)
Comment (ondratu) [reply spec/openapi.yaml:689]: We want to show nice (display) path for user on File detail. And i'm not sure how difficult is, to compute that display_path from previews request in file listing. May be display_path can be optional, which actually is. 

Please figure out @Majkluss and @Lowentwickler

File: spec/openapi.yaml (hunk 6 above)
Comment (vorner) [reply spec/openapi.yaml:653]: Mostly, what I'm trying to say, can you pick only one that's mandatory?

File: spec/openapi.yaml (hunk 5 above)
Comment (vorner) [reply spec/openapi.yaml:689]: Doesn't the server already have all the long path segments of the folders above? Is it really relying on the printer doing all that? 😇

**Comment by Lowentwickler** [comment]
@ondratu we combine path on the fly, currently from names, but I will add also display_path, made from display names

File: spec/openapi.yaml (hunk 5 above)
Comment (Lowentwickler) [reply spec/openapi.yaml:728]: I construct path manually from display_name. There is no need for display path. And for the operations I use paths from `refs`

**Review by vorner — APPROVED** [review]
I think my concerns were resolved (or lost in the history…)

**Review by ondratu — APPROVED** [review]

**Pull request merged**